token_block.block.inc

```diff
@@ -18,7 +18,14 @@ class TokenBlock extends Block {
    * {@inheritdoc}
    */
   function getTitle() {
-    return !empty($this->settings['title']) ? check_plain($this->settings['title']) : t('Token block');
+    if (!empty($this->settings['title'])) {
+      $entity_type = $this->childDelta;
+      $entity = $this->contexts[$entity_type]->data;
+      $data = array($entity_type => $entity);
+      return check_plain(token_replace($this->settings['title'], $data));
+    } else {
```

Comment:
...also, coding standards here (else needs to be on the next line - see https://docs.backdropcms.org/php-standards#control-structures):
```suggestion
    }
    else {
```